src/routes/groupGraphController/GroupGraphController.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { useEffect, useState } from 'react';
+
+import { Button } from 'components/Button/Button';
+import GroupGraph from 'components/GroupGraph/GroupGraph';
+import LeaderboardTeacher from 'components/LeaderboardTeacher/LeaderboardTeacher';
+
+import { groupOptions } from 'routes/group/groupOptions';
+
+import { useIndex } from 'hooks/useIndex';
+import { useNavigate, useLocation } from 'react-router-dom';
+
+import { GroupGraphType } from 'types/GroupGraph/GroupGraphType';
+import { Leaderboard } from 'types/Leaderboard/Leaderboard';
+
+import { leaderboardData } from './leaderboardDummyData';
+import style from './GroupGraphController.module.css';
+
+function getDummyGraphData(
+  isLeaderboard: boolean
+): (GroupGraphType | Leaderboard)[] {
+  if (isLeaderboard) {
+    return leaderboardData;
+  }
+  const data = Array.from({ length: 10 }, () => {
+    const title = 'Modulo de aprendizaje';
+    const value = Math.floor(Math.random() * 101);
+    const id = Math.floor(Math.random() * 101);
+    return { title, value, id };
+  });
+  return data;
+}
+
+export default function GroupGraphController() {
+  const location = useLocation();
+  const navigate = useNavigate();
+
+  const [graphData, setGraphData] = useState<(GroupGraphType | Leaderboard)[]>(
+    []
+  );
+  const { index, max, next, prev, setMaxIndex } = useIndex({
+    initial: parseInt(location.state.id_graph as string, 10) || 0,
```

Comment:
id_graph could be null if the user access to the url path directly from the browser. Provide a safeguard for this

Reply:
I see that if is not defined, it will not take 0. I'll add the safeguard